addon/components/dialog.ts

```diff
@@ -52,6 +52,26 @@ export default class DialogComponent extends Component<Args> {
     }
   );
 
+  lockWindowScroll = modifier(() => {
+    // Opt-out of some other dialog already locked scrolling
+    if (this.dialogStackProvider.dialogIsOpen) {
+      return;
+    }
+
+    let overflow = this.$portalRoot.style.overflow;
+    let paddingRight = this.$portalRoot.style.paddingRight;
+
+    let scrollbarWidth = window.innerWidth - this.$portalRoot.clientWidth;
```

Comment:
can we add a comment here about how this is to prevent layout jitter when the scrollback is hidden due to overflow: hidden being set.
You and I both know why this is here, but I'd like stuff like this to be documented in code to lower the barrier to entry and to enhance understand for broader contributions, ya know? <3

Reply:
Yeah, that's a great idea! Good thinking.